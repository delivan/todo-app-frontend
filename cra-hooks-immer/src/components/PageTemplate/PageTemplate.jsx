import React from 'react';
import classNames from 'classnames/bind';

import styles from './PageTemplate.scss';

const cx = classNames.bind(styles);

export default function PageTemplate({ children }) {
  return (
    <div className={cx('page-template')}>
      <h1>Todo App Frontend</h1>
      <p>skills: CRA(with React Hooks), Immer.js </p>
      <div className={cx('content')}>
        {children}
      </div>
    </div>
  );
};