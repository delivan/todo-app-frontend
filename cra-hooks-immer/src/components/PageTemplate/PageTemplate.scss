.page-template {
  margin: 3rem auto 3rem auto;
  width: 600px;
  background: white;
  padding-top: 2rem;

  @media(max-width: 768px) {
    margin: 1rem;
    width: calc(100% - 2rem);
  }

  h1 {
    text-align: center;
    font-size: 4rem;
    font-weight: 300;
    margin: 0;
  }

  p {
    text-align: center;
    color: palevioletred;
  }

  .content {
    margin-top: 2rem;
  }
}