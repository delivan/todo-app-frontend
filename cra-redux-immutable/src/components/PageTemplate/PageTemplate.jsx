import React from 'react';
import styles from './PageTemplate.scss';
import classNames from 'classnames/bind';

const cx = classNames.bind(styles);

function PageTemplate({ children }) {
  return (
    <div className={cx('page-template')}>
      <h1>Todo App Frontend</h1>
      <div className={cx('content')}>
        {children}
      </div>
    </div>
  );
};

export default PageTemplate;