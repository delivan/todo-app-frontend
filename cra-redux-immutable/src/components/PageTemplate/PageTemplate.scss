.page-template {
  margin: 3rem auto 0 auto;
  width: 600px;
  background: white;
  padding-top: 2rem;

  @media(max-width: 768px) {
    margin-top: 1rem;
    width: calc(100% - 2rem);
  }

  h1 {
    text-align: center;
    font-size: 4rem;
    font-weight: 300;
    margin: 0;
  }

  .content {
    margin-top: 2rem;
  }
}