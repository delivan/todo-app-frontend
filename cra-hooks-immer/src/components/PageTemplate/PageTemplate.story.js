import React from "react";
import { storiesOf } from "@storybook/react";

import PageTemplate from "./PageTemplate";

storiesOf("PageTemplate", module).add("default", () => (
  <PageTemplate>Children...</PageTemplate>
));
