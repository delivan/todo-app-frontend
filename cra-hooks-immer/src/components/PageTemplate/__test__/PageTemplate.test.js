import React from 'react';
import {create} from 'react-test-renderer'; 
import PageTemplate from '../PageTemplate';

it('renders correctly', () => {
  let element = create(<PageTemplate>test</PageTemplate>);
  expect(element.toJSON()).toMatchSnapshot();

  element.update(<PageTemplate>test2</PageTemplate>);
  expect(element.toJSON()).toMatchSnapshot();
});