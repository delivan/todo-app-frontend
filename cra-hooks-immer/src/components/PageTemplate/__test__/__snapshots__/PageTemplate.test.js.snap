// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="page-template"
>
  <h1>
    Todo App Frontend
  </h1>
  <p>
    skills: CRA(with React Hooks), Immer.js 
  </p>
  <div
    className="content"
  >
    test
  </div>
</div>
`;

exports[`renders correctly 2`] = `
<div
  className="page-template"
>
  <h1>
    Todo App Frontend
  </h1>
  <p>
    skills: CRA(with React Hooks), Immer.js 
  </p>
  <div
    className="content"
  >
    test2
  </div>
</div>
`;
